openedx/core/djangoapps/django_comment_common/comment_client/course.py

```diff
@@ -0,0 +1,41 @@
+# pylint: disable=missing-docstring
+"""Provides base Commentable model class"""
+from __future__ import annotations
+
+from typing import Dict
+
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.core.djangoapps.django_comment_common.comment_client import settings
+from openedx.core.djangoapps.django_comment_common.comment_client.utils import perform_request
+
+
+def get_course_commentable_counts(course_key: CourseKey) -> Dict[str, Dict[str, int]]:
+    """
+    Get stats about the count of different types of threads for each commentable (topic).
+
+    Args:
+        course_key (str|CourseKey): course key for which stats are needed.
+
+    Returns:
+        A mapping of topic ids to the number of question and discussion type posts in them.
+
+        e.g.
+            {
+                "general": { "discussion": 22, "question": 15 },
+                "topic-1": { "discussion": 2, "question": 1 },
```

Comment:
@xitij2000 
Question: can you explain these counts? 

what is 22 is referring for `general.discussion`?  is the total number of discussions inside the general topic?

Reply:
Yes, this means there are 22 posts of discussion type in the topic with the id "general".